tests/pyright/type_adapter.py

```diff
@@ -0,0 +1,25 @@
+from typing import Annotated
+
+from typing_extensions import assert_type
+
+from pydantic import TypeAdapter
+
+ta1 = TypeAdapter(int)
+assert_type(ta1, TypeAdapter[int])
+
+assert_type(ta1.validate_python('1'), int)
+ta1.dump_python(1)
+ta1.dump_python('1')  # pyright: ignore[reportArgumentType]
+ta1.dump_json(1)
+ta1.dump_json('1')  # pyright: ignore[reportArgumentType]
+
+# The following use cases require PEP 747: TypeExpr:
```

Comment:
Nice. Will be great to have this as a reference to update when that PEP is formally adopted 👍